### PR TITLE
Refine size-your-shards wording

### PR DESCRIPTION
**Pull request by DaveCTurner**
Clarify that the limits in the docs are absolute maxima that will avoid
things just breaking but won't necessarily give great performance.

### CODE REVIEW

**Comment by DaveCTurner** [comment]
Preview: https://elasticsearch_89081.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/size-your-shards.html

**Review by jakommo — APPROVED** [review]
Thanks a lot @DaveCTurner :heart: This makes it much more clear now !

**Review by original-brownbear — APPROVED** [review]

File: docs/reference/how-to/size-your-shards.asciidoc [hunk 1]
```diff
@@ -175,17 +175,25 @@ index                                 prirep shard store
...
 The number of indices a master node can manage is proportional to its heap
 size. The exact amount of heap memory needed for each index depends on various
 factors such as the size of the mapping and the number of shards per index.
 
-As a general rule of thumb, you should aim for 3000 indices or fewer per GB of
-heap on master nodes. For example, if your cluster contains 12000 indices then
-each dedicated master node should have at least 4GB of heap. For non-dedicated
-master nodes, the same rule holds and should be added to the heap requirements
-of the other roles of each node.
+As a general rule of thumb, you should have fewer than 3000 indices per GB of
```
Comment (henningandersen) [inline docs/reference/how-to/size-your-shards.asciidoc:184]: Can we also reverse the sentence here to derive the GB from number of indices?

**Review by henningandersen — COMMENTED** [review]
Looks good though I have a few more wishes.

File: docs/reference/how-to/size-your-shards.asciidoc [hunk 2]
```diff
@@ -175,17 +175,24 @@ index                                 prirep shard store
 
 [discrete]
 [[shard-count-recommendation]]
-==== Aim for 3000 indices or fewer per GB of heap memory on each master node
+==== Aim for fewer than 3000 indices per GB of heap memory on each master node
```
[the remark this answers is not in the document]
Comment (DaveCTurner) [reply docs/reference/how-to/size-your-shards.asciidoc:178]: I prefer this order for a couple of reasons:

1. IMO it's more common that the user knows the size of their nodes and is trying to work out a suitable sharding and retention strategy to match. For instance on Cloud there's only a few different possible master heap sizes.

2. We want to phrase this as a bound, not a target, and I find it less natural to say "aim for more than 1GB of heap per 3000 indices".

[the remark this answers is not in the document]
Comment (henningandersen) [reply docs/reference/how-to/size-your-shards.asciidoc:178]: About 1, this seems to be the wrong way around and I'd like to encourage that in the heading.

About 2, we can perhaps call it:

```suggestion
==== Masters should have at least 1 GB of heap per 3000 indices.
```

**Review by henningandersen — APPROVED** [review]
LGTM though I'd still like the heading change. Added a suggestion, but do not want to hold back the other additional text.

**Pull request merged**